### DISCUSSION TITLE
question: compile

### DISCUSSION THREAD

**Discussion by ghost** [icsharpcode/ILSpy#3305] [General]
hi

I have a qeustion as a beginner.
Is it possible to compile a decompiled exe again after changing some code?
How to do this?

**Comment by greenozon** [comment]
Usually this is called Reverse Engineering and is illegal activity
besides you are doing this action on our own bits

**Comment by greenozon** [comment]
steps to recompile:
click on Assembly in the left-hand tree view
menu File - Save COde (or just CTRL + S)
wait until it finishes
Click on "Open Explorer" button
double click on .csproj file
it'll open up in your installed VS (if any)
build
in case of luck - success, 
otherwise need to go one by one raised issues and fix it by hands

**Comment by ghost** [comment]
Thanks for your answer!

Is it possible a developer of an EXE file can, on purpose, remove several dependencies out of the EXE so that decompiling is impossible (because you don't have everything)?
Or is it ALWAYS possible to decompile an entire EXE (regardless of the errors)
